### PR TITLE
fix(vscode): keep config order in vscode extension project dropdown

### PR DESCRIPTION
**Pull request by mxschmitt**
Relates https://github.com/microsoft/playwright/issues/30936

Test plan: Cover it on the VSCode extension side once this is released under @next.

### CODE REVIEW

File: packages/playwright/src/runner/loadUtils.ts [hunk 1]
```diff
@@ -67,14 +67,12 @@ export async function collectProjectsAndTestFiles(testRun: TestRun, doNotRunTest
...
   for (const [project, type] of projectClosure) {
     if (type === 'dependency') {
-      filesToRunByProject.delete(project);
       const treatProjectAsEmpty = doNotRunTestsOutsideProjectFilter && !filteredProjects.includes(project);
       const files = treatProjectAsEmpty ? [] : allFilesForProject.get(project) || await collectFilesForProject(project, fsCache);
       filesToRunByProject.set(project, files);
     }
   }
 
-  testRun.projects = [...filesToRunByProject.keys()];
```
Comment (mxschmitt) [inline packages/playwright/src/runner/loadUtils.ts:77]: drive-by: remove unused field.

File: packages/playwright/src/runner/loadUtils.ts [hunk 2]
```diff
@@ -67,14 +67,12 @@ export async function collectProjectsAndTestFiles(testRun: TestRun, doNotRunTest
   const projectClosure = buildProjectsClosure([...filesToRunByProject.keys()]);
   for (const [project, type] of projectClosure) {
     if (type === 'dependency') {
-      filesToRunByProject.delete(project);
```
Comment (mxschmitt) [inline packages/playwright/src/runner/loadUtils.ts:70]: if you do: delete/set it looses the order.
if you do: set, it keeps the order in JavaScript.

**Review by dgozman — APPROVED** [review]
